pkg/termination/handler.go

```diff
@@ -90,5 +92,17 @@ func (h *handler) run(ctx context.Context, wg *sync.WaitGroup) error {
 
 // getMachineForNodeName finds the Machine associated with the Node name given
 func (h *handler) getMachineForNode(ctx context.Context) (*machinev1.Machine, error) {
+	machineList := &machinev1.MachineList{}
+	err := h.client.List(ctx, machineList)
```

Comment:
Do you think would it worth it to create a manager that handles the client and informer so we can index by nodeRef?
We can probably include that later if we want.

https://github.com/kubernetes-sigs/controller-runtime/blob/a7c8a93c1cf395974911ef0ece977a7d27b2bf4b/pkg/manager/manager.go#L320-L335
https://github.com/kubernetes-sigs/controller-runtime/blob/f7e674bbb6c9c122753681a15e7f57c483a5c121/pkg/cache/cache.go#L110-L117

Reply:
I don't know if we need a full blown manager, but a `DelegatingClient` with the cache as you've suggested could definitely be good.

I do wonder whether that's overkill though, since this program only calls the API twice, once to list, and once to delete, so the only benefit to adding the cache is on that list

That said, I am now also thinking about whether we should `Get` the machine before `Deleting`, not sure if the resource version needs to be up to date for a `Delete`, will check